### DISCUSSION TITLE
Multiple JSON Schema

### DISCUSSION THREAD

**Discussion by jl-beast** [kubeapps/kubeapps#5994] [Ideas]
I am creating a Helm chart for an application that requires a database. The application can be run in two ways: either with a remote connection string or with a local file path. The remote connection string requires more variables to be defined in one .yaml file, while the local file path has its own unique variables.

To make it easy for users to deploy the application, I need to provide two different ways to run it. However, duplicating my Helm chart for each option is not ideal.

One idea to solve this is to add a new feature for KubeApps that would allow the user to select a *.schema.json file from a dropdown menu when creating the chart. This would search for all *.schema.json files inside the Helm chart directory and enable users to load the corresponding *.yaml file with default values for the selected option. This new feature would offer users multiple ways to deploy the application defined by the creator.

I do believe that this may not be feasible at this time, due to linting, installing, and other helm commands not supporting the ability to provide alternative linters for feature completeness of this concept, though there may be some ways to get around this. But I wanted to post this anyway as a user case for something that would improve reusability.

*Edit: was going through the issues, is this somewhat supported with the new 2.6.3 version? I am currently on 2.6.0 and the custom alternatives might be what I am looking for. I would still want alternative linting/schemas though... that might not be supported"

**Comment by absoludity** [comment]
Hi @jl-beast . Yeah, part of what you're after is in 2.6.3, but as you've noted, it's specifically for supporting multiple **value** files, not schemas. But we do plan to continue that work and support multiple schema files too (and eventually form schemas), but we're lacking resources on the team right now to do so.

That said, isn't it possible to have two mutually exclusive options on a chart and validation that ensures they are chosen correctly? Hmm, I see in our Kubeapps chart itself, we're enforcing such situations not via a schema validation, but via a `fail` message that does not allow the install... not sure if that would work for you? See 

https://github.com/vmware-tanzu/kubeapps/blob/475d9ad6bfc90d46a31d49ab9924c472cf4d70bd/chart/kubeapps/templates/_helpers.tpl#L227

**Comment by absoludity** [comment]
Did you check some of the bitnami charts? If it's possible to set the schema for mutually exclusive option validation, they'll be doing it.
